docs/actions-go.md

```diff
@@ -17,67 +17,185 @@
+  // encode the result back in json
+  return msg
+}
+```
 
-func main() {
-    //program receives one argument: the JSON object as a string
-    arg := os.Args[1]
+You can deploy it with just:
 
-    // unmarshal the string to a JSON object
-    var obj map[string]interface{}
-    json.Unmarshal([]byte(arg), &obj)
+```
+wsk action create hello-go hello.go
+```
 
-    // can optionally log to stdout (or stderr)
-    fmt.Println("hello Go action")
+You can also have multiple source files in an action, packages and vendor folders.
 
-    name, ok := obj["name"].(string)
-    if !ok { name = "Stranger" }
+## Deployment
+
+The runtime `actionloop-golang-v1.11` accepts:
+
+- executable binaries in Linux ELF executable compiled for the AMD64 architecture
+- zip files containing a binary executable named `exec` at the top level, again a Linux ELF executable compiled for the AMD64 architecture
+- a single file source code in Go language, that will be compiled
+- a zip file not containing in the top level a binary file `exec`, it will be interpreted as a collection of zip files, and compiled
+
+## Using packages and vendor folder
+
+When you deploy a zip file, you can:
+
+- have all your functions in the `main` package
+- have some functions placed in some packages, like `hello`
+- have some third party dependencies you want to include in your sources
+
+If all your functions are in the main package, just place all your sources in the top level of your zip file.
+
+### Use a package folder
+
+If some functions belongs to a package, like `hello/`, you need to be careful with the layout of your sources, especially if you use editors like [VcCode](#vscode), and make. The layout recommended is the following:
 
-    // last line of stdout is the result JSON object as a string
-    msg := map[string]string{"msg": ("Hello, " + name + "!")}
-    res, _ := json.Marshal(msg)
-    fmt.Println(string(res))
-}
 ```
+golang-main-package/
+- Makefile
+- src/
+   - main.go
+   - main_test.go
+   - hello/
+       - hello.go
+       - hello_test.go
+```
+
+For running tests, editing without errors with package resolution, you need to use a `src` folder, place the sources that belongs to the main package in the `src` and place sources of your package in the `src/hello` folder.
+
+You should import it your subpackage with `import "hello"`.
+Note this means if you want to compile locally you have to set your `GOPATH` to parent directory of your `src` directory. If you use VSCode, you need to enable the `go.inferGopath` option.
+
+When you send the sources, you will have to zip the content of the `src` folder, *not* the main directory. For example:
 
-Save the code above to a file `sample.go` and cross compile it for OpenWhisk.
-The executable must be called `exec`.
-```bash
-GOOS=linux GOARCH=amd64 go build -o exec
-zip exec.zip exec
-wsk action create helloGo --native exec.zip
+```
+cd src
+zip -r ../hello.zip *
+cd ..
+wsk action create hello2-go hello.zip --kind go:1.11
```

Comment:
I think the action names can just be hello to avoid the unnecessary hyphe. Easier to read. Maybe instead of hello2 it’s hellozip